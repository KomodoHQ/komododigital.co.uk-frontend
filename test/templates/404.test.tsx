import React from 'react';
import FourOhFour from '../../src/templates/404';
import renderer from 'react-test-renderer';
import GatsbyData from '../../gatsby-config';

test('404 template is rendered', () => {
  const hocProps = {
    htmlAst: {
      type: 'element',
      tagName: 'span',
      properties: {},
      children: [
        {
          type: 'text',
          value: 'Komodo Digital',
        },
      ],
    },
    data: {
      site: GatsbyData,
      logo: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      }
    },
  };

  const component = renderer.create(<FourOhFour {...hocProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
