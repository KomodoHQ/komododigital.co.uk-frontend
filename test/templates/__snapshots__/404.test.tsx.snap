// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 template is rendered 1`] = `
Array [
  <noscript>
    You have Javascript disabled. While it isn't needed, it will make your experience nicer.
  </noscript>,
  <div
    className="komodoGridWrapper navigationBar "
    style={
      Object {
        "background": "#000000",
      }
    }
  >
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": 29,
            "overflow": "hidden",
            "position": "relative",
            "width": 170,
          }
        }
      >
        
        <picture>
          
          <source
            srcSet=""
          />
          <img
            alt=""
            height={29}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
            width={170}
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img width=\\"170\\" height=\\"29\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
    <div
      className="hamburger"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <img
          height={64}
          src={Object {}}
          style={
            Object {
              "height": 64,
              "marginTop": 10,
              "width": 64,
            }
          }
          width={64}
        />
      </a>
    </div>
    <ul
      className="Menu"
      style={
        Object {
          "background": "#000000",
        }
      }
    >
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/case-studies"
          onClick={[Function]}
        >
          Case Studies
        </a>
      </li>
      <li>
        <a
          href="/blog-list"
          onClick={[Function]}
        >
          Insights
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="main"
    style={
      Object {
        "background": "linear-gradient(to bottom,  80px,#EAEAEA 80px)",
      }
    }
  >
    <h1>
      #404 Page Not Found
    </h1>
    <div>
      <span>
        Komodo Digital
      </span>
    </div>
  </div>,
  <footer
    className="komodoGridWrapper"
  >
    <div
      className="komodo"
    >
      <img
        alt="Komodo - People Friendly Software"
        src={Object {}}
      />
    </div>
    <div
      className="address"
    >
      <span>
        Address
      </span>
      <address>
        Komodo Digital,
        <br />
        63 Westgate Rd,
        <br />
        Newcastle upon Tyne,
        <br />
        NE1 1SG
      </address>
    </div>
    <div
      className="sitemap"
    >
      <span>
        Sitemap
      </span>
      <ul>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Case Studies
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Insights
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Careers
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/komodohq"
          >
            Twitter
          </a>
        </li>
        <li>
          <a
            href="https://facebook.com/komodohq"
          >
            Facebook
          </a>
        </li>
        <li>
          <a
            href="https://linkedin.com/komodohq"
          >
            LinkedIn
          </a>
        </li>
      </ul>
    </div>
  </footer>,
  <footer
    className="komodoGridWrapper legal-footer"
  >
    <div
      className="company"
    >
      <span>
        Komodo Digital Copyright Â© 2018
      </span>
      <br />
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Privacy Policy
      </a>
       | 
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Terms and Conditions
      </a>
    </div>
    <div
      className="contact"
    >
      Get in touch on:
      <br />
      <a
        href="/tel:0191 228 6555"
        onClick={[Function]}
      >
        0191 228 6555
      </a>
    </div>
  </footer>,
  <noscript>
    <img
      alt=""
      height="1"
      src="https://dc.ads.linkedin.com/collect/?pid=377068&fmt=gif"
      width="1"
    />
  </noscript>,
]
`;
